File: fds-dashboard/src/packages/fds-service/src/alert-fds-parameter/dto/alert-fds-parameter.dto.ts
import { BasePageableResponse } from "@/packages/base/src/response/base.pageable.response";
export type AlertFdsParameterDto = {
  authSeqNo: string;
  cardNo: string;
  memberBankAcq: string;
  merchantName: string;
  mid: string;
  tid: string;
  rrn: string;
  issuer: string;
  authDate: string;
  authTime: string;
  authAmount: number;
  traceNo: string;
  messageTypeId: string;
  authSaleType: string;
  authIntnRspnCd: string;
  reasonContents: string;
  installmentCount: number;
  switchBrand: string;
  posEntryModeDetail: string;
  cardTypeCode: string;
  onusCode: string;
  eciValue: string;
  approvalCode: string;
  pgName: string;
  pgType: string;
  issuerMemberNo: string;
  businessType: string;
  channel: string;
  issuerCountry: string;
  parameterValues: string;
  actionType: string | null;
  fraudType: string | null;
  fraudPodType: string | null;
  fraudNote: string | null;
  confirmedBy: string | null;
  confirmedDateTime: Date | null;
  remindNote: string | null;
  remindDate: Date | null;
  assignedTo: string | null;
  assignedDateTime: Date | null;
  authDateTime: Date | null;
  uniqueId: string | null;

  activityId: string | null;
  id: string | null;
};

export type AlertFdsParameterPageableDto = {
  content: AlertFdsParameterDto[];
} & BasePageableResponse;

File: fds-backend/fds-service/src/main/java/com/yokke/fdsservice/alert_fds_parameter/AlertFdsParameterDto.java
package com.yokke.fdsservice.alert_fds_parameter;

import com.yokke.fdsservice.alert_fds_parameter_history.AlertFdsParameterHistoryDto;
import jakarta.persistence.Column;
import lombok.*;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.time.OffsetDateTime;
import java.util.List;

/**
 * DTO for {@link AlertFdsParameter}
 */
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class AlertFdsParameterDto implements Serializable {
    String authSeqNo;
    String cardNo;
    String memberBankAcq;
    String merchantName;
    String mid;
    String tid;
    String rrn;
    String issuer;
    String authDate;
    String authTime;
    BigDecimal authAmount;
    String traceNo;
    String messageTypeId;
    String authSaleType;
    String authIntnRspnCd;
    String reasonContents;
    Integer installmentCount;
    String switchBrand;
    String posEntryModeDetail;
    String cardTypeCode;
    String onusCode;
    String eciValue;
    String approvalCode;
    String pgName;
    String pgType;
    String issuerMemberNo;
    String businessType;
    String channel;
    String issuerCountry;
    String parameterValues;
    String actionType;
     String fraudType;

    String fraudPodType;
    String fraudNote;
    String confirmedBy;
    LocalDateTime confirmedDateTime;
    String remindNote;
    LocalDateTime remindDate;
    String assignedTo;
    LocalDateTime assignedDateTime;


    OffsetDateTime authDateTime;
    String uniqueId;
    String notes;

     String activity;
     String id;


    List<AlertFdsParameterHistoryDto> alertFdsParameterHistory;
}